### DISCUSSION TITLE
Best way to import your profiles on Anne2?

### DISCUSSION THREAD

**Discussion by ezra-sg** [astronautr/60p-keyboard-layout#3] [Q&A]
Hi! first off, thank you SO much for sharing these ideas, i'm also a developer and the ideas you have around ergonomic/efficiency are amazing. love the brackets right above the spacebar especially 🤓  

question: when I go into the obinskit software, go to the layout tab in the left nav, and click the import button next to "Library", to import one of your macOS json files, it seems to import in an "off by one" fashion—the `Default` layer is mapped to your `Fn1` layer keys, the `Fn1` layer is a mapped to your `Fn2` keys, and the `Fn2` layer is blank (see attached pics). I was wondering if i'm attempting to import them in the wrong way, or if perhaps something changed in a recent version and you need to re-export the jsons. 

![Screen Shot 2021-03-02 at 1 18 17 PM](https://user-images.githubusercontent.com/11837820/109695363-cefe2500-7b59-11eb-9bb8-ecb2ad534cf0.png)
![Screen Shot 2021-03-02 at 1 18 23 PM](https://user-images.githubusercontent.com/11837820/109695364-cf96bb80-7b59-11eb-9643-1f1156e38c39.png)
![Screen Shot 2021-03-02 at 1 18 28 PM](https://user-images.githubusercontent.com/11837820/109695366-cf96bb80-7b59-11eb-9b63-c20337c188da.png)
![Screen Shot 2021-03-02 at 1 18 32 PM](https://user-images.githubusercontent.com/11837820/109695367-cf96bb80-7b59-11eb-8f52-d15595569bd8.png)

I'm on version 1.1.5 of obinskit, for reference.

**Comment by ezra-sg** [comment]
ah okay i should have just inspected the JSONs before commenting. creating a PR now to add mapping for version 1.1.5, it seems they have switched to a 1 based index rather than 0, i.e. from the Default being `data.layer0` to `data.layer1`, and bumped the other layers up accordingly

**Reply by astronautr** [reply]
Hello! Thank you for your feedback and contribution :) Looking forward to your PR

**Reply by ezra-sg** [reply]
ah thanks for the quick reply! i pretty much immediately ran into issues here haha, it seems that editing a JSON and then re-importing it doesnt work. my hunch is that it's because of a mismatch of the CRC field, which seems to be a sort of UUID for the file. so for instance, if i simply take one of your JSONs, update the keys from `data.layer0` to `data.layer1` and try to import, ObinsKit throws an error. have you encountered this / are you aware of a workaround? if not, i'll just manually recreate and export the mappings 😄

**Reply by ezra-sg** [reply]
any chance you know your ObinsKit version and your actual keyboard firmware versions? i'm on the latest of both since i just bought the keyboard today haha, obinskit v1.1.5, KB firmware 1.14.6. it would be cool if i could track down what obinskit is doing under the hood; i also noticed that when i create a fresh layout and export it, my json has `"model":"5"` whereas yours has `"model":"3"` 🤔

**Reply by astronautr** [reply]
Sorry, I hear about this issue for the first time and don't know any workarounds except recreating the whole keymap manually

**Reply by ezra-sg** [reply]
ok cool, thank you! it'll be an interesting puzzle for me haha

**Reply by astronautr** [reply]
Yeah, my keyboard is relatively old I've bought it about two years ago and it has been broken for six months already. I no longer use this board and don't have any obins software installed for this long, sorry can't help you here

**Reply by ezra-sg** [reply]
ah understood. here's that PR!
https://github.com/astronautr/60p-keyboard-layout/pull/4
